.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDatePeriodParser.cs

```diff
@@ -1257,67 +1262,131 @@ private DateTimeResolutionResult ParseYear(string text, DateObject referenceDate
         }
 
         // Parse entities that are made up by two time points
-        private DateTimeResolutionResult MergeTwoTimePoints(string text, DateObject referenceDate)
+        private DateTimeResolutionResult MergeTwoTimePoints(string text, DateObject referenceDate)
+        {
+            var ret = new DateTimeResolutionResult();
+
+            var er = this.config.DateExtractor.Extract(text, referenceDate);
+            if (er.Count < 2)
+            {
+                er = this.config.DateExtractor.Extract(this.config.TokenBeforeDate + text, referenceDate);
+                if (er.Count < 2)
+                {
+                    return ret;
+                }
+
+                er[0].Start -= this.config.TokenBeforeDate.Length;
+                er[1].Start -= this.config.TokenBeforeDate.Length;
+            }
+
+            var match = this.config.WeekWithWeekDayRangeRegex.Match(text);
+            string weekPrefix = null;
+            if (match.Success)
+            {
+                weekPrefix = match.Groups["week"].ToString();
+            }
+
+            if (!string.IsNullOrEmpty(weekPrefix))
+            {
+                er[0].Text = weekPrefix + " " + er[0].Text;
+                er[1].Text = weekPrefix + " " + er[1].Text;
+            }
+
+            var dateContext = GetYearContext(er[0].Text, er[1].Text, text);
+
+            var pr1 = this.config.DateParser.Parse(er[0], referenceDate);
+            var pr2 = this.config.DateParser.Parse(er[1], referenceDate);
+            if (pr1.Value == null || pr2.Value == null)
+            {
+                return ret;
+            }
+
+            pr1 = dateContext.ProcessDateEntityParsingResult(pr1);
+            pr2 = dateContext.ProcessDateEntityParsingResult(pr2);
+
+            ret.SubDateTimeEntities = new List<object> { pr1, pr2 };
+
+            DateObject futureBegin = (DateObject)((DateTimeResolutionResult)pr1.Value).FutureValue,
+                futureEnd = (DateObject)((DateTimeResolutionResult)pr2.Value).FutureValue;
+
+            DateObject pastBegin = (DateObject)((DateTimeResolutionResult)pr1.Value).PastValue,
+                pastEnd = (DateObject)((DateTimeResolutionResult)pr2.Value).PastValue;
+
+            if (futureBegin > futureEnd)
+            {
+                futureBegin = pastBegin;
+            }
+
+            if (pastEnd < pastBegin)
+            {
+                pastEnd = futureEnd;
+            }
+
+            ret.Timex = $"({pr1.TimexStr},{pr2.TimexStr},P{(futureEnd - futureBegin).TotalDays}D)";
+            ret.FutureValue = new Tuple<DateObject, DateObject>(futureBegin, futureEnd);
+            ret.PastValue = new Tuple<DateObject, DateObject>(pastBegin, pastEnd);
+            ret.Success = true;
+
+            return ret;
+        }
+
+        // Parse entities that are made up by two time points with now
+        private DateTimeResolutionResult MergeTwoTimePointsWithNow(string text, DateObject referenceDate)
```

Comment:
Why do we need two methods? Also, are the other changes just line ending changes? If so, please revert them to make review easier.

Reply:
In the first method, we will re_extract "date" from  "text", "now" is a Date time, so it would not be extracted, could not solve the issue

Reply:
There is also a change in the extractor extracting "now" in this PR, no? Why do we need it again parsing?

Reply:
And it could be part of the same parsing method as a special case, couldn't it?